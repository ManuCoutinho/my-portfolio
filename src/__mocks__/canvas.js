HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
