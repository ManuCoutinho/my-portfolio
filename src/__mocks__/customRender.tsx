import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from './providers/DataContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
