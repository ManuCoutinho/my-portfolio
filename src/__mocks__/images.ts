export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
