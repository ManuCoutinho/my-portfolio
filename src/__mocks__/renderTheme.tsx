import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
