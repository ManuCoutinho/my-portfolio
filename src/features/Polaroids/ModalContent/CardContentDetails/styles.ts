import styled, { css } from 'styled-components'

export const BoxDetails = styled.div`
	${({ theme }) => css`
		width: 100%;
		display: flex;
		flex-direction: row;
		align-items: baseline;
		justify-self: stretch;
		flex-wrap: nowrap;
		padding: 0.5rem 0.8rem;
		gap: 1rem;
		border-radius: ${theme.radius.sm};
		box-shadow: 1px 1px 5px 2px rgba(0, 0, 0, 0.15);
		transition: color 0.3s ease-in-out;
		&:last-of-type {
			align-items: center;
		}
		@media only screen and (max-width: 80em) {
			padding: 0.85rem;
		}
	`}
`
export const Topic = styled.dt`
	${({ theme }) => css`
		font-weight: ${theme.fontWeight.semibold};
		font-family: ${theme.font.regular};
		color: ${theme.colors.info};
		font-size: ${theme.fontSize.lg};
		white-space: pre;
		@media only screen and (max-width: 80em) {
			white-space: pre-wrap;
			font-size: 1.2rem;
			width: 15ch;
		}
	`}
`
export const IconLink = styled.a`
	${({ theme }) => css`
		font-size: ${theme.fontSize.lg};
		padding: 0.5rem;
		&:hover {
			color: ${theme.colors.active};
			transition: color 0.3s ease-in-out;
		}
	`}
`
