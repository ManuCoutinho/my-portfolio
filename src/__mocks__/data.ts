import { mockImg } from './images'
export default {
	description: 'a belezura de test',
	site: 'http://www.somesite.com',
	repo: 'http://www.somerepo.com',
	styles: 'styles',
	api: 'mock',
	framework: 'next',
	name: 'project',
	tools: ['tools'],
	id: '81j37ldd-3k344',
	doc: 'http://www.storybook.com',
	coverDefault: '/assets/preview.png',
	coverAlt: '/public/assets/preview.png',
	alt: 'mock img',
	img: mockImg,
	route: '/test'
}
