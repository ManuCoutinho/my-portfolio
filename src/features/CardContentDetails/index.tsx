import Link from 'next/link'
import { Icon } from '@iconify-icon/react'
import { StackBox } from 'components/StackBox'
import { Text } from 'components/Text'
import { DataProps } from 'features/Card/types'
import * as Styled from './styles'
import icons from 'constants/icons'

const CardContentDetails: React.FC<DataProps> = ({ ...props }) => {
  return (
    <StackBox id='details' direction='column' gap={1} px='1.5rem'>
      <Styled.BoxDetails>
        <Styled.Topic>Descrição: </Styled.Topic>
        <Text as='dd'>{props.description}</Text>
      </Styled.BoxDetails>
      {props.framework && (
        <Styled.BoxDetails>
          <Styled.Topic>Framework: </Styled.Topic>
          <Text as='dd'>{props.framework}</Text>
        </Styled.BoxDetails>
      )}
      <Styled.BoxDetails>
        <Styled.Topic>Linguagens e Ferramentas: </Styled.Topic>
        <Text as='dd'>{props.tools?.join(' / ')}</Text>
      </Styled.BoxDetails>
      <Styled.BoxDetails>
        <Styled.Topic>Estilização: </Styled.Topic>
        <Text as='dd'>{props.styles}</Text>
      </Styled.BoxDetails>
      {props.api && (
        <Styled.BoxDetails>
          <Styled.Topic>API: </Styled.Topic>
          <Text as='dd'>{props.api}</Text>
        </Styled.BoxDetails>
      )}
      <Styled.BoxDetails as='dd'>
        <Styled.Topic>Disponível em: </Styled.Topic>
        <Link href={props.repo} passHref>
          <Styled.IconLink
            target='_blank'
            rel='noopener noreferrer'
            title='repositorio no github'
          >
            <Icon icon={icons.github} style={{ fontSize: '1.8rem' }} />
          </Styled.IconLink>
        </Link>
        {props.site != 'Em breve' && (
          <Link href={props.site} passHref>
            <Styled.IconLink
              target='_blank'
              rel='noopener noreferrer'
              title={`acessar o ${props.name}`}
            >
              <Icon icon={icons.link_external} />
            </Styled.IconLink>
          </Link>
        )}
        {props.doc && (
          <Link href={props.doc} passHref>
            <Styled.IconLink
              target='_blank'
              rel='noopener noreferrer'
              title={`documentation for ${props.name}`}
            >
              <Icon icon={icons.docs} />
            </Styled.IconLink>
          </Link>
        )}
      </Styled.BoxDetails>
    </StackBox>
  )
}

export default CardContentDetails
