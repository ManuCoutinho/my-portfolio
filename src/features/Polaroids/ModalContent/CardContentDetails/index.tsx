import Link from 'next/link'
import { Icon } from '@iconify-icon/react'

import { StackBox } from 'components/StackBox'
import { Text } from 'components/Text'

import icons from 'constants/icons'
import { DataType } from '@models/data'
import * as Styled from './styles'

const CardContentDetails: React.FC<DataType> = ({ ...props }) => {
	return (
		<StackBox id='details' direction='column' gap={1} px='1.5rem' as='dl'>
			<Styled.BoxDetails>
				<Styled.Topic aria-label='description'>Descrição: </Styled.Topic>
				<Text as='dd' aria-label={props.description}>
					{props.description}
				</Text>
			</Styled.BoxDetails>
			{props.framework && (
				<Styled.BoxDetails>
					<Styled.Topic aria-label={'Framework'}>Framework: </Styled.Topic>
					<Text as='dd' aria-label={props.framework}>
						{props.framework}
					</Text>
				</Styled.BoxDetails>
			)}
			<Styled.BoxDetails>
				<Styled.Topic aria-label={'tools'}>
					Linguagens e Ferramentas:{' '}
				</Styled.Topic>
				<Text as='dd' aria-label={props.tools}>
					{props.tools?.join(' / ')}
				</Text>
			</Styled.BoxDetails>
			<Styled.BoxDetails>
				<Styled.Topic aria-label={'style'}>Estilização: </Styled.Topic>
				<Text as='dd' aria-label={props.styles}>
					{props.styles}
				</Text>
			</Styled.BoxDetails>
			{props.api && (
				<Styled.BoxDetails>
					<Styled.Topic aria-label={'api'}>API: </Styled.Topic>
					<Text as='dd' aria-label={props.api}>
						{props.api}
					</Text>
				</Styled.BoxDetails>
			)}
			<Styled.BoxDetails>
				<Styled.Topic aria-label={'available'}>Disponível em: </Styled.Topic>
				<Link href={props.repo} passHref>
					<Styled.IconLink
						target='_blank'
						rel='noopener noreferrer'
						title='repositorio no github'
					>
						<Icon icon={icons.github} style={{ fontSize: '1.8rem' }} />
					</Styled.IconLink>
				</Link>
				{props.site != 'Em breve' && (
					<Link href={props.site} passHref>
						<Styled.IconLink
							target='_blank'
							rel='noopener noreferrer'
							title={`acessar o ${props.name}`}
						>
							<Icon icon={icons.link_external} />
						</Styled.IconLink>
					</Link>
				)}
				{props.doc && (
					<Link href={props.doc} passHref>
						<Styled.IconLink
							target='_blank'
							rel='noopener noreferrer'
							title={`documentation for ${props.name}`}
						>
							<Icon icon={icons.docs} />
						</Styled.IconLink>
					</Link>
				)}
			</Styled.BoxDetails>
		</StackBox>
	)
}

export default CardContentDetails
