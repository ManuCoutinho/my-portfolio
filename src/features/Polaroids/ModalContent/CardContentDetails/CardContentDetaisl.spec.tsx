import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { setupIntersectionObserverMock } from '__mocks__/intersectionObserver'
import CardContentDetails from '.'
import data from '__mocks__/data'

describe('<CardContentDetails />', () => {
	beforeEach(() => setupIntersectionObserverMock())

	it('should render the content modal correctly', async () => {
		render(<CardContentDetails {...data} />)
		expect(screen.getByText(/a belezura de test/i)).toBeInTheDocument()
		expect(
			screen.getByRole('term', { name: /description/i })
		).toBeInTheDocument()
		const projectLink = screen.getByRole('link', { name: /acessar o project/i })
		expect(projectLink).toHaveAttribute('href', 'http://www.somesite.com')
		expect(projectLink.firstChild).toHaveAttribute(
			'icon',
			'codicon:link-external'
		)
		expect(screen.getByTestId('stackbox')).toBeInTheDocument()
	})

	it('should match to snapshot', async () => {
		const { container } = render(<CardContentDetails {...data} />)
		expect(container).toMatchSnapshot()
	})
})
