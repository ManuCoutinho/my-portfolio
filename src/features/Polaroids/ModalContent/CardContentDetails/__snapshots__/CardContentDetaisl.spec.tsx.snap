// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContentDetails /> should match to snapshot 1`] = `
<div>
  <dl
    class="sc-ce9dac95-0 bkDPty"
    data-testid="stackbox"
    direction="column"
    gap="1"
    id="details"
    px="1.5rem"
  >
    <div
      class="sc-bc329a75-0 gbjSaM"
    >
      <dt
        aria-label="description"
        class="sc-bc329a75-1 ifNFYe"
      >
        Descrição: 
      </dt>
      <dd
        class="sc-f89260d2-0 eyKYlB"
      >
        a belezura de test
      </dd>
    </div>
    <div
      class="sc-bc329a75-0 gbjSaM"
    >
      <dt
        aria-label="Framework"
        class="sc-bc329a75-1 ifNFYe"
      >
        Framework: 
      </dt>
      <dd
        class="sc-f89260d2-0 eyKYlB"
      >
        next
      </dd>
    </div>
    <div
      class="sc-bc329a75-0 gbjSaM"
    >
      <dt
        aria-label="tools"
        class="sc-bc329a75-1 ifNFYe"
      >
        Linguagens e Ferramentas:
         
      </dt>
      <dd
        class="sc-f89260d2-0 eyKYlB"
      >
        tools
      </dd>
    </div>
    <div
      class="sc-bc329a75-0 gbjSaM"
    >
      <dt
        aria-label="style"
        class="sc-bc329a75-1 ifNFYe"
      >
        Estilização: 
      </dt>
      <dd
        class="sc-f89260d2-0 eyKYlB"
      >
        styles
      </dd>
    </div>
    <div
      class="sc-bc329a75-0 gbjSaM"
    >
      <dt
        aria-label="api"
        class="sc-bc329a75-1 ifNFYe"
      >
        API: 
      </dt>
      <dd
        class="sc-f89260d2-0 eyKYlB"
      >
        mock
      </dd>
    </div>
    <div
      class="sc-bc329a75-0 gbjSaM"
    >
      <dt
        aria-label="available"
        class="sc-bc329a75-1 ifNFYe"
      >
        Disponível em: 
      </dt>
      <a
        class="sc-bc329a75-2 eFEWPK"
        href="http://www.somerepo.com"
        rel="noopener noreferrer"
        target="_blank"
        title="repositorio no github"
      >
        <iconify-icon
          icon="iconoir:github-outline"
          style="font-size: 1.8rem;"
        />
      </a>
      <a
        class="sc-bc329a75-2 eFEWPK"
        href="http://www.somesite.com"
        rel="noopener noreferrer"
        target="_blank"
        title="acessar o project"
      >
        <iconify-icon
          icon="codicon:link-external"
        />
      </a>
      <a
        class="sc-bc329a75-2 eFEWPK"
        href="http://www.storybook.com"
        rel="noopener noreferrer"
        target="_blank"
        title="documentation for project"
      >
        <iconify-icon
          icon="tabler:brand-storybook"
        />
      </a>
    </div>
  </dl>
</div>
`;
